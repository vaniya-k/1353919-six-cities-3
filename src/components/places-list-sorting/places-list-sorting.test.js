import React from 'react';
import renderer from 'react-test-renderer';
import PlacesListSorting from './places-list-sorting.jsx';
import {createStore} from "redux";
import {Provider} from "react-redux";

const mockReducerSortedByPopular = () => {
  const state = {
    offers: {
      activeSortType: 0
    }
  };

  return state;
};

const mockStoreSortedByPopular = createStore(mockReducerSortedByPopular);

const mockReducerSortedByPriceLowToHigh = () => {
  const state = {
    offers: {
      activeSortType: 1
    }
  };

  return state;
};

const mockStoreSortedByPriceLowToHigh = createStore(mockReducerSortedByPriceLowToHigh);

const mockReducerSortedByPriceHighToLow = () => {
  const state = {
    offers: {
      activeSortType: 2
    }
  };

  return state;
};

const mockStoreSortedByPriceHighToLow = createStore(mockReducerSortedByPriceHighToLow);

const mockReducerSortedByTopRated = () => {
  const state = {
    offers: {
      activeSortType: 3
    }
  };

  return state;
};

const mockStoreSortedByTopRated = createStore(mockReducerSortedByTopRated);

const mock = jest.fn();

it(`<PlacesListSorting/> is sorted with popular first`, () => {
  const tree = renderer
    .create(<Provider store={mockStoreSortedByPopular}><PlacesListSorting isOpened={false} onDropdownToggle={mock}/></Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`<PlacesListSorting/> is sorted by price from low to high`, () => {
  const tree = renderer
    .create(<Provider store={mockStoreSortedByPriceLowToHigh}><PlacesListSorting isOpened={false} onDropdownToggle={mock}/></Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`<PlacesListSorting/> is sorted by price from high to low`, () => {
  const tree = renderer
    .create(<Provider store={mockStoreSortedByPriceHighToLow}><PlacesListSorting isOpened={false} onDropdownToggle={mock}/></Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`<PlacesListSorting/> is sorted with top rated first`, () => {
  const tree = renderer
    .create(<Provider store={mockStoreSortedByTopRated}><PlacesListSorting isOpened={false} onDropdownToggle={mock}/></Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
