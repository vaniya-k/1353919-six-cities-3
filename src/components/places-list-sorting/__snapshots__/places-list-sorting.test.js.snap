// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlacesListSorting/> is sorted by price from high to low 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by 
  </span>
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex="0"
  >
    Price: high to low
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        href="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom null"
  >
    <li
      className="places__option"
      onClick={[Function]}
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option places__option--active"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option"
      onClick={[Function]}
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
</form>
`;

exports[`<PlacesListSorting/> is sorted by price from low to high 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by 
  </span>
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex="0"
  >
    Price: low to high
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        href="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom null"
  >
    <li
      className="places__option"
      onClick={[Function]}
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option places__option--active"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option"
      onClick={[Function]}
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
</form>
`;

exports[`<PlacesListSorting/> is sorted with popular first 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by 
  </span>
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex="0"
  >
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        href="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom null"
  >
    <li
      className="places__option places__option--active"
      onClick={[Function]}
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option"
      onClick={[Function]}
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
</form>
`;

exports[`<PlacesListSorting/> is sorted with top rated first 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by 
  </span>
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex="0"
  >
    Top rated first
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        href="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom null"
  >
    <li
      className="places__option"
      onClick={[Function]}
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option places__option--active"
      onClick={[Function]}
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
</form>
`;
